packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -237,6 +237,7 @@ const SceneView = ({
     : parentHeaderBack;
 
   const isRemovePrevented = preventedRoutes[route.key]?.preventRemove;
+  const isFocused = navigation.isFocused();
```

Comment:
`navigation.isFocused()` shouldn't be used during render. The appropriate one is `useIsFocused` but it can't be used here since this isn't a screen. To handle it properly, we need to pass `focused` prop to `SceneView` from the parent.

Reply:
Passed `focused` from the parent using following check: `const isFocused = state.index === index;`